src/managed/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.Netstandard2_0.cs

```diff
@@ -0,0 +1,628 @@
+                    case DependencyContextStrings.TargetsPropertyName:
+                        targets = ReadTargets(ref reader);
+                        break;
+                    case DependencyContextStrings.LibrariesPropertyName:
+                        libraryStubs = ReadLibraries(ref reader);
+                        break;
+                    case DependencyContextStrings.RuntimesPropertyName:
+                        runtimeFallbacks = ReadRuntimes(ref reader);
+                        break;
+                    default:
+                        reader.Skip();
+                        break;
+                }
+            }
+
+            if (compilationOptions == null)
+            {
+                compilationOptions = CompilationOptions.Default;
+            }
+
+            Target runtimeTarget = SelectRuntimeTarget(targets, runtimeTargetName);
+            runtimeTargetName = runtimeTarget?.Name;
+
+            if (runtimeTargetName != null)
+            {
+                var separatorIndex = runtimeTargetName.IndexOf(DependencyContextStrings.VersionSeparator);
+                if (separatorIndex > -1 && separatorIndex < runtimeTargetName.Length)
+                {
+                    runtime = runtimeTargetName.Substring(separatorIndex + 1);
+                    framework = runtimeTargetName.Substring(0, separatorIndex);
+                    isPortable = false;
+                }
+                else
+                {
+                    framework = runtimeTargetName;
+                }
+            }
+
+            Target compileTarget = null;
+
+            var ridlessTarget = targets.FirstOrDefault(t => !IsRuntimeTarget(t.Name));
+            if (ridlessTarget != null)
+            {
+                compileTarget = ridlessTarget;
+                if (runtimeTarget == null)
+                {
+                    runtimeTarget = compileTarget;
+                    framework = ridlessTarget.Name;
+                }
+            }
+
+            if (runtimeTarget == null)
+            {
+                throw new FormatException("No runtime target found");
+            }
+
+            return new DependencyContext(
+                new TargetInfo(framework, runtime, runtimeSignature, isPortable),
+                compilationOptions,
+                CreateLibraries(compileTarget?.Libraries, false, libraryStubs).Cast<CompilationLibrary>().ToArray(),
+                CreateLibraries(runtimeTarget.Libraries, true, libraryStubs).Cast<RuntimeLibrary>().ToArray(),
+                runtimeFallbacks ?? Enumerable.Empty<RuntimeFallbacks>());
+        }
+
+        private void ReadRuntimeTarget(ref Utf8JsonReader reader, out string runtimeTargetName, out string runtimeSignature)
+        {
+            runtimeTargetName = null;
+            runtimeSignature = null;
+
+            reader.ReadStartObject();
+
+            while (reader.TryReadStringProperty(out string propertyName, out string propertyValue))
```

Comment:
Same question about extracting strings and whether spans could be used first, then materialized into strings only when needed.  Of course, maybe perf isn't a big factor here, in which case clarity wins.  I'll stop asking about this.

Reply:
> Of course, maybe perf isn't a big factor here, in which case clarity wins. I'll stop asking about this.

Based on the Json.NET usage within this library, I got the sense that perf isn't a big factor. We can still optimize the implementation for netstandard2.0 given the new library though.

https://github.com/dotnet/core-setup/issues/5034

Resolving as out of scope for this PR.